githubscript/src/githubscript/script.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+""" Github main script
+"""
+import logging
+import os
+
+from githubscript.github import release
+from githubscript.release_config import get_release_config
+from scriptworker_client.client import sync_main
+from scriptworker_client.exceptions import ClientError, TaskVerificationError
+from scriptworker_client.utils import get_single_item_from_sequence
+
+log = logging.getLogger(__name__)
+
+
+async def async_main(config, task):
+    project_config = _get_project_config(config, task)
+    release_config = get_release_config(project_config, task["payload"], config)
+    contact_server = bool(config.get("contact_server"))
+
+    _log_warning_forewords(contact_server)
+
+    action = task["payload"]["action"]
+    if action == "release":
+        await release(release_config)
+    else:
+        raise NotImplementedError(f'Action "{action}" is not supported')
+
+    log.info("Done!")
+
+
+def _get_project_config(config, task):
+    try:
+        project_configs = config["github_projects"]
+    except KeyError:
+        raise ClientError('"github_projects" is not part of the configuration')
+
+    config_scopes = project_configs.keys()
+    config_scope = get_single_item_from_sequence(
+        sequence=task["scopes"],
+        condition=lambda task_scope: task_scope in config_scopes,
+        ErrorClass=TaskVerificationError,
+        no_item_error_message="Task contains no scope supported by this instance",
+        too_many_item_error_message="Task contains too many scopes supported by this instance. Please provide only a single one",
+    )
+
+    return project_configs[config_scope]
+
+
+def _log_warning_forewords(contact_server):
```

Comment:
Not sure I understand the naming of this function. It's probably fine. If we want something more generally useful, we could do something like `check_config()`, with the idea that we may add more sanity checks in the future.

Reply:
Renamed into `_warn_contact_github()`